File: scripts/convert_pdfs.py
import os
import sys
import glob

try:
    import fitz  # PyMuPDF
except ImportError:
    print("PyMuPDF not installed. Please run: pip install pymupdf")
    sys.exit(1)

def convert_pdf_to_jpg(pdf_path):
    print(f"Processing: {pdf_path}")
    try:
        doc = fitz.open(pdf_path)
        base_path = os.path.splitext(pdf_path)[0]
        
        # Determine if multi-page
        num_pages = len(doc)
        for i in range(num_pages):
            page = doc.load_page(i)
            # Render at 300 DPI roughly
            pix = page.get_pixmap(matrix=fitz.Matrix(2, 2))
            
            if num_pages == 1:
                out_path = f"{base_path}.png"
            else:
                out_path = f"{base_path}_page_{i+1}.png"
                
            pix.save(out_path)
            print(f"Saved: {out_path}")
            
        doc.close()
        
        # Delete original PDF after successful conversion
        os.remove(pdf_path)
        print(f"Deleted original PDF: {pdf_path}")
        
    except Exception as e:
        print(f"Failed to convert {pdf_path}: {e}")

if __name__ == "__main__":
    project_works_dir = os.path.join("public", "PROJECT WORKS")
    if not os.path.exists(project_works_dir):
        print(f"Directory not found: {project_works_dir}")
        sys.exit(1)
        
    # Find all PDFs recursively
    pdf_files = glob.glob(os.path.join(project_works_dir, "**", "*.pdf"), recursive=True)
    print(f"Found {len(pdf_files)} PDF files to convert.")
    
    for pdf in pdf_files:
        convert_pdf_to_jpg(pdf)
        
    print("Done converting PDFs.")

File: scripts/convertPdfs.mjs
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';
import pdf2img from 'pdf-img-convert';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const projectsDir = path.join(__dirname, '..', 'public', 'PROJECT WORKS');

async function convertPdfs(dir) {
  const files = fs.readdirSync(dir);

  for (const file of files) {
    const fullPath = path.join(dir, file);
    if (fs.statSync(fullPath).isDirectory()) {
      await convertPdfs(fullPath);
    } else if (file.toLowerCase().endsWith('.pdf')) {
      console.log(`Converting: ${fullPath}`);
      try {
        const outputImages = await pdf2img.convert(fullPath, {
          width: 2000,
          page_numbers: [1] // Convert only the first page or all? Let's just do page 1 assuming it's single page or they want the first page.
        });

        // The user asked to convert pdf into jpg. Architectural works might have multiple pages.
        // Let's convert all pages.
        const outputImagesAll = await pdf2img.convert(fullPath, {
          width: 2000,
        });

        if (outputImagesAll.length === 1) {
            const outPath = fullPath.replace(/\.pdf$/i, '.jpg');
            fs.writeFileSync(outPath, outputImagesAll[0]);
            console.log(`Saved: ${outPath}`);
        } else {
            for (let i = 0; i < outputImagesAll.length; i++) {
                const outPath = fullPath.replace(/\.pdf$/i, `_page_${i + 1}.jpg`);
                fs.writeFileSync(outPath, outputImagesAll[i]);
                console.log(`Saved: ${outPath}`);
            }
        }
        
        // Delete original PDF
        fs.unlinkSync(fullPath);
        console.log(`Deleted original: ${fullPath}`);
      } catch (err) {
        console.error(`Failed to convert ${fullPath}:`, err);
      }
    }
  }
}

async function run() {
    console.log("Starting PDF conversion...");
    await convertPdfs(projectsDir);
    console.log("Done.");
}

run();
